kbase-extension/static/kbase/js/widgets/appWidgets2/fieldWidgetMicro.js

```diff
@@ -0,0 +1,561 @@
+            component.events.attachEvents(document.body);
+        }
+
+        function setWarning(warning) {
+            var component = buildInputMessage({
+                title: 'Warning',
+                type: 'warning',
+                message: warning.message,
+                id: warning.id
+            });
+            places.messagePanel.classList.remove('hidden');
+            places.message.innerHTML = component.content;
+            places.message.classList.add('-warning');
+            component.events.attachEvents(document.body);
+        }
+
+
+        function clearError() {
+            places.field.classList.remove('-error');
+            places.field.classList.remove('-warning');
+            places.message.classList.remove('-error');
+            places.message.classList.remove('-warning');
+            places.message.innerHTML = '';
+            places.messagePanel.classList.add('hidden');
+        }
+
+        function hideError() {
+            places.field.classList.remove('-error');
+            places.messagePanel.classList.add('hidden');
+            places.feedbackIndicator.className = '';
+        }
+
+        function feedbackNone() {
+            places.feedbackIndicator.className = '';
+            places.feedbackIndicator.classList.add('hidden');
+        }
+
+        function feedbackOk() {
+            places.feedbackIndicator.className = '';
+            places.feedbackIndicator.setAttribute('title', 'input is ok');
+            places.feedbackIndicator.classList.remove('hidden');
+        }
+
+        function feedbackRequired() {
+            places.feedbackIndicator.className = 'kb-app-parameter-required-glyph fa fa-arrow-left';
+            places.feedbackIndicator.setAttribute('title', 'required field');
+        }
+
+        function feedbackError() {
+            places.feedbackIndicator.className = 'kb-app-parameter-required-glyph fa fa-ban';
+        }
+
+        function rawSpec(spec) {
+            var specText = JSON.stringify(spec, false, 3),
+                fixedSpec = specText.replace(/</g, '&lt;').replace(/>/g, '&gt;');
+            return pre({ class: 'prettyprint lang-json', style: { fontSize: '80%' } }, fixedSpec);
+        }
+
+        function parameterInfoContent(spec) {
+            return div({ style: { padding: '0px' } }, [
+                div({ style: { fontWeight: 'bold' } }, spec.ui.label),
+                div({ style: { fontStyle: 'italic' } }, spec.id),
+                div({ style: { fontSize: '80%' } }, spec.ui.description)
+            ]);
+        }
+
+        function parameterInfoTypeRules(spec) {
+            switch (spec.data.type) {
+            case 'float':
+                return [
+                    tr([th('Min'), td(spec.data.constraints.min)]), // update this in the spec
+                    tr([th('Max'), td(spec.data.constraints.max)])
+                ];
+            case 'int':
+                // just for now ...
```

Comment:
Are 'int' and 'float' expected to be treated the same here? If so, should just merge into the same case.